utils/s3/aws_error.cc

```diff
@@ -150,7 +154,9 @@ aws_error aws_error::from_system_error(const std::system_error& system_error) {
     case GNUTLS_E_BAD_COOKIE: // as per RFC6347 section-4.2.1 client should retry
         return {aws_error_type::NETWORK_CONNECTION, system_error.code().message(), retryable::yes};
     default:
-        return {aws_error_type::NETWORK_CONNECTION, system_error.code().message(), retryable::no};
+        return {aws_error_type::NETWORK_CONNECTION,
+                format("Issuing non-retryable network related error. Message: {}, code: {}", system_error.code().message(), system_error.code().value()),
+                retryable::no};
```

Comment:
I'm confused. Should this return retryable::yes now?

Can we test those error paths?

Reply:
"Non-retryable system error occurred"